### PR TITLE
deprecate the imagestream.spec.DockerImageRepository field

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@soltysh @smarterclayton ptal.

**Comment by jim-minter** [comment]
@bparees hack/update-generated-swagger-descriptions.sh needed

File: pkg/image/apis/image/v1/types.go [hunk 1]
```diff
@@ -195,7 +197,8 @@ type TagReference struct {
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 	// Annotations associated with images using this tag
 	Annotations map[string]string `json:"annotations" protobuf:"bytes,2,rep,name=annotations"`
-	// From is a reference to an image stream tag or image stream this tag should track
+	// Optional; if specified, a reference to another image that this tag should point to. Valid values
```
Comment (jim-minter) [inline pkg/image/apis/image/v1/types.go]: Question - should the docstrings for other fields match ../types.go too?

Comment (bparees) [reply pkg/image/apis/image/v1/types.go]: probably, will update.

**Comment by bparees** [comment]
@jim-minter fixed
@openshift/api-review ptal

File: pkg/image/apis/image/v1/types.go [hunk 2]
```diff
@@ -173,6 +173,8 @@ type ImageStreamSpec struct {
 	// lookupPolicy controls how other resources reference images within this namespace.
 	LookupPolicy ImageLookupPolicy `json:"lookupPolicy,omitempty" protobuf:"bytes,3,opt,name=lookupPolicy"`
 	// dockerImageRepository is optional, if specified this stream is backed by a Docker repository on this server
+	// Deprecated: This field is deprecated as of v3.7 and will be removed in a future release
```
Comment (smarterclayton) [inline pkg/image/apis/image/v1/types.go]: Needs a period after this sentence.

Comment (bparees) [reply pkg/image/apis/image/v1/types.go]: fixed.

**Comment by bparees** [comment]
/retest

Ben Parees | OpenShift

**Comment by bparees** [comment]
/retest

Ben Parees | OpenShift

On Sep 7, 2017 00:37, "OpenShift CI Robot" <notifications@github.com> wrote:

@bparees <https://github.com/bparees>: The following tests *failed*, say
/retest to rerun them all:
Test name Commit Details Rerun command
ci/openshift-jenkins/extended_builds 34dc649
<https://github.com/openshift/origin/commit/34dc649d45d6b0579b323ff832cdc09ceaa3cd72>
link
<https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/16181/test_pull_request_origin_extended_builds/195/>
/test
extended_builds
ci/openshift-jenkins/extended_conformance_gce 745e72b
<https://github.com/openshift/origin/commit/745e72bdba663e5c9bdd2c09c428ff3549a91544>
link
<https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/16181/test_pull_request_origin_extended_conformance_gce/7407/>
/test
extended_conformance_gce
ci/openshift-jenkins/verify 745e72b
<https://github.com/openshift/origin/commit/745e72bdba663e5c9bdd2c09c428ff3549a91544>
link
<https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/16181/test_pull_request_origin_verify/2185/>
/test
verify

Full PR test history <https://openshift-gce-devel.appspot.com/pr/16181>. Your
PR dashboard <https://openshift-gce-devel.appspot.com/pr/bparees>. Please
help us cut down on flakes by linking to
<https://github.com/kubernetes/community/blob/master/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests>
an open issue
<https://github.com/openshift/origin/issues?q=is:issue+is:open> when you
hit one in your PR.

Instructions for interacting with me using PR comments are available here
<https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md>.
If you have questions or suggestions related to my behavior, please file an
issue against the kubernetes/test-infra
<https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
repository. I understand the commands that are listed here
<https://github.com/kubernetes/test-infra/blob/master/commands.md>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/16181#issuecomment-327682737>,
or mute
the thread

**Comment by bparees** [comment]
/retest

Ben Parees | OpenShift

**Comment by bparees** [comment]
going to assume @smarterclayton's /approve qualifies as a 
/lgtm

and api-approval.

**Comment by smarterclayton** [comment]
/approve

for api

**Pull request merged**